httpx/client_test.go

```diff
@@ -43,6 +44,41 @@ func (c *FailingClient) Do(*http.Request) (*http.Response, error) {
 	}, nil
 }
 
+type MetricsInstrument struct {
+	metrics.Metrics
+	timerMock metrics.Timer
+}
+
+func (mi *MetricsInstrument) Timer(name string, tags ...metrics.Tag) metrics.Timer {
+	return mi.timerMock
+}
+
+type TimerMock struct {
```

Comment:
this is a spy, not a mock; see https://github.com/gonzaloserrano/go-nzalo/tree/master/code-review#tests cc/ @ronnylt who was specially "picky" about this